### ISSUE TITLE
Add thrown objects (i.e. Redirect, HttpError) to @sveltejs/kit package exports

### ISSUE THREAD

**Issue by nicholas-wilcox** [sveltejs/kit#10062]
### Describe the problem

I would like to write unit tests for my server hooks. Specifically, I want to add test cases that make assertions about SvelteKit's "control" objects that get thrown by `redirect`, `error`, and `fail`.

What I can do now, using Vitest as it comes preconfigured by `create-svelte`, is something like the following:

```typescript
// src/hooks.server.test.ts
import type { RequestEvent } from '@sveltejs/kit';
import { describe, expect, it, vi } from 'vitest';
import * as hooks from './hooks.server';
import { mockRequestEvent } from '$lib/mocks/request'; // My own module

describe('redirect hook', () => {
  it('throws redirect', async () => {
    event = mockRequestEvent() // My own function
    const resolve = vi.fn(async (_event: RequestEvent) => new Response());
    const test = async () => hooks.redirectHook({ event, resolve });
    await expect(test).rejects.toThrowError();
    expect(resolve).toHaveBeenCalledTimes(0);
  });
});
```

However, any usage of the optional argument of `toThrowError()` will result in an error, because the objects that are thrown aren't actually errors.

What I would like to to is write custom tests using `expect.extend` that use the `instanceof` operator to check that the thrown object is the correct type.

The weird thing about this is that TypeScript lets me write an import statement that _looks_ like it imports the `Redirect` class, not the type. Unfortunately, the symbol evaluates to `undefined` during runtime, which makes code like `actual instanceof Redirect` or `Redirect.prototype` throw runtime errors.

In other words, this is valid code for a fresh SvelteKit application made with `create-svelte`:
```typescript
// src/hooks.server.ts
import type { Handle, RequestEvent } from '@sveltejs/kit';
import { error, redirect, Redirect } from '@sveltejs/kit';

export const handle = (async ({ event, resolve }) => {
  console.log(Redirect); // prints "undefined"
  return await resolve(event);
}) satisfies Handle;
```

However, trying to print `Redirect.prototype` throws a `TypeError`.

### Describe the proposed solution

Export `Redirect`, `HttpError`, and `ActionFailure` in https://github.com/sveltejs/kit/blob/master/packages/kit/src/exports/index.js.

Adding an export statement manually in `node_modules/@sveltejs/kit/src/exports/index.js` prevents the errors described above.

### Alternatives considered

I only want access to these objects for testing purpose, so I can also use Vitest's mocking features to create a mock for `@sveltejs/kit`. However, this solution requires me to redeclare the classes in my own module:

```typescript
// src/setup-test.ts -- Added as a setup file in vite.config.ts
import { vi } from 'vitest';
import { Redirect, HttpError, error, redirect } from '$lib/mocks/control';

vi.doMock('@sveltejs/kit', () => ({ Redirect, redirect }));

expect.extend({
  toThrowRedirect(actual: Redirect, status: number, location?: string) {
    if (!(actual instanceof Redirect)) {
      return {
        pass: false,
        message: () => 'Object is not Redirect type',
      };
    }

    // Individual checks on status and location omitted for brevity
  }
})
```

Now, the thrown objects will have accessible prototypes, but they will be from the classes declared in my own module.

### Importance

nice to have

### Additional Information

I'm not sure if the quirk that prevents a type-check error qualifies as a bug.

Also, because we aren't supposed to catch these objects in our application code, I can't think of a reason that anyone would want to access `Redirect.prototype` outside of testing.

**Comment by nicholas-wilcox** [comment]
I'm aware of the "[grotesque](https://github.com/sveltejs/kit/blob/c8310d7bedcd95905671224cc928f40a2bcc9386/packages/kit/src/runtime/control.js#L48)" and "[shameful](https://github.com/sveltejs/kit/blob/c8310d7bedcd95905671224cc928f40a2bcc9386/packages/kit/src/exports/vite/dev/index.js#L346)" hack that relates to these objects. However, I don't fully understand what's going on here.

I don't think the way this module is loaded has anything to do with the issues I'm experiences, but I also don't know if my proposed solution will break the hack.

**Comment by rebasecase** [comment]
Bump bump

**Comment by nicholas-wilcox** [comment]
I am coming back to close this issue because the `isRedirect` and `isHttpError` functions, introduced in SvelteKit v2, resolve the problems I encountered back when I created this issue.

I've created a demo repository that explains my current approach to unit testing server hooks and making assertions about SvelteKit's thrown objects more broadly. https://github.com/nicholas-wilcox/sveltekit-server-hook-unit-tests

**Comment by rebasecase** [comment]
Could use .rejects.toSatisfy(isHttpError)?

**Comment by nicholas-wilcox** [comment]
@rebasecase Yes, that is a much simpler solution than my approach. It should cover many if not most use cases.

You could also drill down into the error data with the following code:
```ts
it('returns a 500 status', async () => {
  let { event, resolve } = mockHandleParams();
  await expect(errorHook({ event, resolve })).rejects.toSatisfy((e) => {
    return isHttpError(e) && e.status === 500;
  });
});
```

However, at this point I would still recommend a [custom matcher](https://vitest.dev/guide/extending-matchers.html) that could encapsulate the logic and standardize how the thrown objects are checked while reducing code duplication. Additionally, if the above test fails because a different status was used, the failed test output isn't as clear (in my opinion, at least).

**Comment by elliott-with-the-longest-name-on-github** [comment]
@nicholas-wilcox `isHttpError` actually accepts a second argument for the specific status code you're looking for, so you can simplify from `isHttpError(e) && e.status === 500` to `isHttpError(e, 500)`
